json/src/test/java/com/basistech/rosette/dm/json/array/JsonTest.java

```diff
@@ -207,19 +209,22 @@ public void oneWithEverything() {
         araMaBuilder.lemma("lemma");
         araMaBuilder.partOfSpeech("pos");
         araMaBuilder.raw("raw");
+        araMaBuilder.tagSet(TagSet.BT_ARABIC);
         tokenBuilder.addAnalysis(araMaBuilder.build());
 
         HanMorphoAnalysis.Builder hanMaBuilder = new HanMorphoAnalysis.Builder();
         hanMaBuilder.addReading("proust");
         hanMaBuilder.lemma("lemma");
         hanMaBuilder.partOfSpeech("pos");
+        // test null tag set
         tokenBuilder.addAnalysis(hanMaBuilder.build());
 
         KoreanMorphoAnalysis.Builder korMaBuilder = new KoreanMorphoAnalysis.Builder();
         korMaBuilder.addMorpheme("m1", "t1");
         korMaBuilder.addMorpheme("m2", "t2");
         korMaBuilder.partOfSpeech("korean");
         korMaBuilder.lemma("koreanLemma");
+        korMaBuilder.tagSet(TagSet.BT_KOREAN);
```

Comment:
By calling `tagSet` in this test you’ve shown that nothing crashes, but you should also put something in `roundTrip` to make sure tag sets are properly deserialized.

Reply:
It checks that the `token` it gets after deserializing is the same as the `token` it creates, which means that it checks that the analyses are equal too, and I changed the `equals` method for `MorphoAnalysis` to check that the tag sets are equal; is that sufficient?

Reply:
You are right: that is sufficient.